PRIOR_U_F_COV : 0.5
PRIOR_U_M1_COV: 0.05
PRIOR_U_M2_COV: 0.05
PRIOR_U_M3_COV: 0.05

# Good for nominal MPC
# PRI_VICON_COV    : 0.001
# PRI_VICON_VEL_COV: 0.001

# CONTROL_P_COV_X      : 0.03
# CONTROL_P_FINAL_COV_X: 0.005
# CONTROL_P_COV_Y      : 0.03
# CONTROL_P_FINAL_COV_Y: 0.005
# CONTROL_P_COV_Z      : 0.03
# CONTROL_P_FINAL_COV_Z: 0.005

# CONTROL_R_COV: 0.03

# CONTROL_V_COV: 3000
# CONTROL_O_COV: 3000

# DYNAMIC_P_COV: 0.001

# OPT_LENS_TRAJ: 20

# SIM_STEPS: 100000

# INPUT_JERK_T: 0.1
# INPUT_JERK_M: 0.05



# method + vel + case_id
LOG_NAME: TGYRO

PRI_VICON_COV    : 0.0001 # 0.20
PRI_VICON_VEL_COV: 0.0001 # 0.05

# PRI_VICON_COV    : 0.20 # 0.20
# PRI_VICON_VEL_COV: 0.05 # 0.05

CONTROL_P_COV_X      : 0.05
CONTROL_P_FINAL_COV_X: 0.03
CONTROL_P_COV_Y      : 0.04
CONTROL_P_FINAL_COV_Y: 0.03
CONTROL_P_COV_Z      : 0.04
CONTROL_P_FINAL_COV_Z: 0.04
CONTROL_R1_COV       : 0.3 #0.40 ~ 0.20
CONTROL_R2_COV       : 0.3
CONTROL_R3_COV       : 0.06
POINT_OBS_SIGMA      : 0.01
CONTROL_V_COV        : 3
CONTROL_O_COV        : 3000000
DYNAMIC_P_COV        : 0.001
DYNAMIC_T_COV        : 0.01

OPT_LENS_TRAJ: 60

SIM_STEPS: 30000

INPUT_JERK_T : 0.1
INPUT_JERK_M : 0.01
INPUT_JERK_M3: 0.01

WINDOW_SIZE: 30

CLF_HIGH: 27
CLF_LOW : 5

G_CLF_HIGH: 3
G_CLF_LOW : -3

CLF_THR  : 1
G_CLF_THR: 0.1

CLF_ALPHA: 1

MAX_ITERS: 10
# distance factor
CBF_ALPHA: 0.9
# vel factor
CBF_BETA: 0.3
# # method + vel + case_id
# LOG_NAME: JECM_R1.2

# PRI_VICON_COV    : 0.20
# PRI_VICON_VEL_COV: 0.05

# CONTROL_P_COV_X      : 0.03
# CONTROL_P_FINAL_COV_X: 0.005
# CONTROL_P_COV_Y      : 0.03
# CONTROL_P_FINAL_COV_Y: 0.005
# CONTROL_P_COV_Z      : 0.01
# CONTROL_P_FINAL_COV_Z: 0.005

# CONTROL_R_COV: 1.2
# # if R_COV is 0.3 (JERK_T = 1, JERK_M = 0.1), the z will converge. WHY???

# CONTROL_V_COV: 3000
# CONTROL_O_COV: 3000

# DYNAMIC_P_COV: 0.001

# OPT_LENS_TRAJ: 20

# SIM_STEPS: 100000

# INPUT_JERK_T: 4
# INPUT_JERK_M: 1

# WINDOW_SIZE: 10





# # method + vel + case_id
# LOG_NAME: SW_JECM_Rev_onlyGNSS_5_N2_

# PRI_VICON_COV    : 0.20
# PRI_VICON_VEL_COV: 0.05

# CONTROL_P_COV_X      : 0.03
# CONTROL_P_FINAL_COV_X: 0.005
# CONTROL_P_COV_Y      : 0.03
# CONTROL_P_FINAL_COV_Y: 0.005
# CONTROL_P_COV_Z      : 0.03
# CONTROL_P_FINAL_COV_Z: 0.005

# CONTROL_R_COV: 1.2

# CONTROL_V_COV: 3000
# CONTROL_O_COV: 3000

# DYNAMIC_P_COV: 0.001

# OPT_LENS_TRAJ: 20

# SIM_STEPS: 100000

# INPUT_JERK_T: 8
# INPUT_JERK_M: 0.5

# WINDOW_SIZE: 10






# # method + vel + case_id
# LOG_NAME: JECM_onlyGNSS_5_N2_

# PRI_VICON_COV    : 0.20
# PRI_VICON_VEL_COV: 0.04

# CONTROL_P_COV_X      : 0.03
# CONTROL_P_FINAL_COV_X: 0.01
# CONTROL_P_COV_Y      : 0.03
# CONTROL_P_FINAL_COV_Y: 0.01
# CONTROL_P_COV_Z      : 0.03
# CONTROL_P_FINAL_COV_Z: 0.01

# CONTROL_R_COV: 0.3

# CONTROL_V_COV: 3000
# CONTROL_O_COV: 3000

# DYNAMIC_P_COV: 0.001

# OPT_LENS_TRAJ: 20

# SIM_STEPS: 100000

# INPUT_JERK_T: 0.1
# INPUT_JERK_M: 0.05





# # method + vel + case_id
# LOG_NAME: JECM_Rev_onlyGNSS_5_1_

# PRI_VICON_COV    : 0.04
# PRI_VICON_VEL_COV: 0.05

# CONTROL_P_COV_X      : 0.03
# CONTROL_P_FINAL_COV_X: 0.03
# CONTROL_P_COV_Y      : 0.03
# CONTROL_P_FINAL_COV_Y: 0.03
# CONTROL_P_COV_Z      : 0.03
# CONTROL_P_FINAL_COV_Z: 0.03

# CONTROL_R_COV: 0.15

# CONTROL_V_COV: 3000
# CONTROL_O_COV: 3000

# DYNAMIC_P_COV: 0.001

# OPT_LENS_TRAJ: 20

# SIM_STEPS: 100000

# INPUT_JERK_T: 4.0
# INPUT_JERK_M: 1




# # method + vel + case_id
# LOG_NAME: MPC_NoNoise_Rev_onlyGNSS_5_1_

# PRI_VICON_COV    : 0.001
# PRI_VICON_VEL_COV: 0.001

# CONTROL_P_COV_X      : 0.03
# CONTROL_P_FINAL_COV_X: 0.005
# CONTROL_P_COV_Y      : 0.03
# CONTROL_P_FINAL_COV_Y: 0.005
# CONTROL_P_COV_Z      : 0.03
# CONTROL_P_FINAL_COV_Z: 0.005

# CONTROL_R_COV: 0.150

# CONTROL_V_COV: 3000
# CONTROL_O_COV: 3000

# DYNAMIC_P_COV: 0.001

# OPT_LENS_TRAJ: 20

# SIM_STEPS: 100000

# INPUT_JERK_T: 0.1
# INPUT_JERK_M: 0.05




# # method + vel + case_id
# LOG_NAME         : MPC_onlyGNSS_5_1_
# PRI_VICON_COV    : 0.001
# PRI_VICON_VEL_COV: 0.001

# CONTROL_P_COV_X      : 0.03
# CONTROL_P_FINAL_COV_X: 0.005
# CONTROL_P_COV_Y      : 0.03
# CONTROL_P_FINAL_COV_Y: 0.005
# CONTROL_P_COV_Z      : 0.03
# CONTROL_P_FINAL_COV_Z: 0.005

# CONTROL_R_COV: 0.03

# CONTROL_V_COV: 3000
# CONTROL_O_COV: 3000

# DYNAMIC_P_COV: 0.001

# OPT_LENS_TRAJ: 20

# SIM_STEPS: 100000

# INPUT_JERK_T: 0.1
# INPUT_JERK_M: 0.05



# LOG_NAME         : JECM_onlyGNSS_5_1_
# PRI_VICON_COV    : 0.20
# PRI_VICON_VEL_COV: 0.05

# CONTROL_P_COV_X      : 0.03
# CONTROL_P_FINAL_COV_X: 0.005
# CONTROL_P_COV_Y      : 0.03
# CONTROL_P_FINAL_COV_Y: 0.005
# CONTROL_P_COV_Z      : 0.03
# CONTROL_P_FINAL_COV_Z: 0.005

# CONTROL_R_COV: 0.3

# CONTROL_V_COV: 3000
# CONTROL_O_COV: 3000

# DYNAMIC_P_COV: 0.001

# OPT_LENS_TRAJ: 20

# SIM_STEPS: 100000

# INPUT_JERK_T: 1
# INPUT_JERK_M: 0.5